docs/types.rst

```diff
@@ -857,13 +859,18 @@ or create a new memory array and copy every element.
         }
     }
 
-.. index:: ! array;literals, !inline;arrays
+.. index:: ! array;literals, ! inline;arrays
+
+Array Literals
+^^^^^^^^^^^^^^
 
-Array Literals / Inline Arrays
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+An array literal is a comma-separated list of one or more expressions, enclosed in square
+brackets (``[...]``). For example ``[1, a, f(3)]``. The elements must have a
```

Comment:
all expressions always have a type.

Reply:
I assume https://github.com/ethereum/solidity/pull/5437#discussion_r replaces this?